ragnarok_packets/src/position.rs

```diff
@@ -1,15 +1,83 @@
-use ragnarok_bytes::{ByteReader, ConversionResult, FromBytes, ToBytes};
+use ragnarok_bytes::{ByteConvertable, ByteReader, ConversionResult, FromBytes, ToBytes};
+
+#[derive(Debug, Copy, Clone, ByteConvertable)]
+#[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
+pub enum Direction {
+    N = 0,
+    NE = 1,
+    E = 2,
+    SE = 3,
+    S = 4,
+    SW = 5,
+    W = 6,
+    NW = 7,
+}
+
+impl From<Direction> for usize {
+    fn from(value: Direction) -> Self {
+        value as usize
+    }
+}
+
+impl From<usize> for Direction {
+    fn from(value: usize) -> Self {
+        let value = value & 7;
+
+        match value {
+            0 => Direction::N,
+            1 => Direction::NE,
+            2 => Direction::E,
+            3 => Direction::SE,
+            4 => Direction::S,
+            5 => Direction::SW,
+            6 => Direction::W,
+            7 => Direction::NW,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl From<&[isize; 2]> for Direction {
```

Comment:
At least only implement the type on the non reference type! De-reference the type before calling the into function.